clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java

```diff
@@ -57,10 +57,12 @@
 
     private final int node;
 
+    private volatile boolean fetchInProgress;
```

Comment:
The class-level javadoc should thoroughly document the thread-safety of this class. And the `sessionPartitions` variable should be a ConcurrentHashMap rather than a LinkedHashMap (as [KAFKA-7280](https://issues.apache.org/jira/browse/KAFKA-7280) shows).

Reply:
Have included the changes in this PR under the thread safety section. To avoid any risky changes, only changes specific to this issue are included in this PR. For changes to other variables, it will be better to create a separate JIRA describing the issue.